client/jetpack-cloud/sections/partner-portal/credit-card-fields/set-as-primary-payment-method.tsx

```diff
@@ -0,0 +1,44 @@
+import styled from '@emotion/styled';
+import { CheckboxControl } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import { useDispatch } from 'react-redux';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+
+const CheckboxWrapper = styled.div`
+	margin-top: 16px;
+`;
+
+export default function SetAsPrimaryPaymentMethod( {
+	isChecked,
+	isDisabled,
+	onChange,
+}: {
+	isChecked: boolean;
+	isDisabled?: boolean;
+	onChange: ( isChecked: boolean ) => void;
+} ): JSX.Element {
+	const translate = useTranslate();
+	const reduxDispatch = useDispatch();
+
+	const handleChangeEvent = ( newIsChecked: boolean ): void => {
+		reduxDispatch(
+			recordTracksEvent( 'calypso_partner_portal_add_credit_card_as_default_toggle', {
+				old_value: isChecked,
+				new_value: newIsChecked,
+			} )
+		);
+		onChange( newIsChecked );
+	};
+
+	return (
+		<CheckboxWrapper>
+			<CheckboxControl
+				className="credit-card-fields__set-as-primary-field"
+				disabled={ isDisabled }
+				checked={ isChecked }
+				onChange={ handleChangeEvent }
+				label={ translate( 'Set as primary payment method' ) }
+			/>
+		</CheckboxWrapper>
```

Comment:
@kallehauge, I was thinking about this one, and from a user perspective, I would like to know why this checkbox is disabled. It seems like it's possible to add your first payment method and not set it as a primary in Stripe, so maybe we should leave this as-is for the moment?

Reply:
@vitozev IMO even though it's possible in Stripe, I think (from a UX perspective) it would make more sense to cover 80% of cases. I figure you could set it up like this:

1. An agency wants to add their first payment method to the portal but forgets to select "primary" (or might not understand what it implies)
2. An agency wants to add two payment method to the portal, but the first card they add should not be used for upcoming invoices

I don't think the second scenario is very common, so I would probably try and go for the "support 80% of cases" rule.
If agencies wants to add multiple cards, they can "just" set the second, third, fourth card as the primary when they add it, so making the first one primary shouldn't have much impact*

_* Perhaps assuming that they don't have queued up invoices that's gone overdue - but I still think the first scenario is the one we should cater towards (reduce friction of us charging the customer by allowing a manual behaviour to incorrectly add their first payment method_ 😄